app/src/main/java/com/naxtlevelofandroiddevelopment/marvelgallery/network/MarvelRepository.kt

```diff
@@ -0,0 +1,28 @@
+package com.naxtlevelofandroiddevelopment.marvelgallery.network
+
+import com.naxtlevelofandroiddevelopment.marvelgallery.model.MarvelCharacter
+import com.naxtlevelofandroiddevelopment.marvelgallery.network.providers.Provider
+import com.naxtlevelofandroiddevelopment.marvelgallery.network.providers.RetrofitApi
+import io.reactivex.Single
+
+interface MarvelRepository {
+
+    fun getCharacters(searchQuery: String?): Single<List<MarvelCharacter>>
+
+    class Impl(val api: MarvelApi) : MarvelRepository {
```

Comment:
Only valid way of doing DI is injecting dependencies using constructor, this way you do not have problems with testing.
For unit test you can use Mockito 2.1+, it is able to stub/mock final classes.

Future proofing for Kotlin Native is a kind of breaking YAGNI.